src/variorum/Intel_DGPU/power_features.c

```diff
@@ -0,0 +1,136 @@
+// Copyright 2019-2022 Lawrence Livermore National Security, LLC and other
+// Variorum Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: MIT
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+
+#include <power_features.h>
+#include <config_architecture.h>
+#include <variorum_error.h>
+#include <variorum_timers.h>
+#include <libapmidg.h>
+
+static unsigned m_total_unit_devices;
+static unsigned m_gpus_per_socket;
+static char m_hostname[1024];
+
+void initAPMIDG(void)
+{
+    int verbose = 0;
+    unsigned m_num_package;
+    apmidg_init(verbose);
+
+    m_total_unit_devices = apmidg_getndevs();
+
+    /* Collect number of packages and GPUs per package */
+    variorum_get_topology(&m_num_package, NULL, NULL);
+    m_gpus_per_socket = m_total_unit_devices / m_num_package;
+
+    /* Save hostname */
+    gethostname(m_hostname, sizeof(m_hostname));
+}
+
+void shutdownAPMIDG(void)
```

Comment:
@slabasan 

This has the same structure bug  with init/finish as the NVIDIA port that we recently fixed with shutdown. For now, we at least need to call shutdownAPMIDG() from the config_architecture.c, otherwise it never gets called. It is much better to contain the init/finish in each individual functions for cleaner code, like I did for the AMD GPU port, but that cleanup can wait if we want to get this merged in sooner.

@kazutomo 
Not sure how much time you have and if this is a lot of work (and if so, let me know and I can give you an initial changes that you can build/test at your end). 

But it may be better and more modular code to do what I did in the AMD GPU port that we recently merged in, see here: https://github.com/LLNL/variorum/blob/dev/src/variorum/AMD_GPU/power_features.c#L28 (and L95 too), even though this is repetitive, it actually contains all the logic in the same file, makes config_<arch> files cleaner (just to detect arch and set pointers), and avoids need for shutdown calls from higher levels,

I know you built off the Nvidia port, and I'm sorry that wasn't a great starting example (but what we had at the time, now we have the AMD example). If we call the init function in the config_intel_dgpu file, we need to call shutdownAPMIDG() at the top-level and include finalize calls there, which is just confusing code fr readers/developers. https://github.com/LLNL/variorum/blob/dev/src/variorum/config_architecture.c#L96 . It also creates dependency between headers that need to be included at that level sometimes.

We need to clean up the Nvidia and Intel msr side eventually too -- we haven't gotten around to it. If we have time, it may be a better thing to fix this before we merge this PR. I can help, just let me know.